runtime/lib/backend/LightningKokkosSimulator.cpp

```diff
@@ -251,32 +255,33 @@ auto LightningKokkosSimulator::Var(ObsIdType obsKey) -> double
     return m.var(*obs);
 }
 
-auto LightningKokkosSimulator::State() -> std::vector<std::complex<double>>
+void LightningKokkosSimulator::State(std::span<std::complex<double>> state)
 {
-    const size_t num_qubits = this->device_sv->getNumQubits();
+    RT_ASSERT(state.size())
+    const size_t num_qubits = this->GetNumQubits();
     const size_t size = Pennylane::Util::exp2(num_qubits);
-    std::vector<std::complex<double>> state;
-    state.reserve(size);
+
+    RT_FAIL_IF(state.size() != size, "Invalid size for the pre-allocated state vector");
 
     for (size_t idx = 0; idx < size; idx++) {
         auto elem_subview = Kokkos::subview(this->device_sv->getData(), idx);
         Kokkos::complex<double> elem_cp;
         Kokkos::deep_copy(elem_cp, elem_subview);
```

Comment:
If this is an actual accelerator device, would it make sense (if there is a way to retrieve all elements)?

Reply:
totally make sense! I was using `Kokkos::subview` for the first draft implementation of this device and totally forgot to revisit this later on. I fixed it in PR #103. Thanks for the suggestion :)